add page argument in getAllMovies serivce

Mi4-04 opened the pull request:
fix service setWatchedMovie and removeMovie

Mi4-04 committed:
add page argument in getAllMovies serivce
fix service setWatchedMovie and removeMovie

Isidora25 commented on src/fav-movies/dto/fav-movies.dto.ts:
```diff
@@ -9,16 +9,16 @@ export class FavMoviesDto {
 
   @Field()
-  originalLanguage: string;
+  original_language: string;
```
У тебя получается цирк с именованием переменных, где-то snake_case, где-то camelCase
Попробуй указать в параметрах для field { name: camelCaseName }, тогда в схему будет попадать все в правильном формате

Mi4-04 committed:
fix fav-movies.dto

Mi4-04 committed:
fix schema

Mi4-04 committed:
add getFavMovies

Isidora25 commented on src/user/user.service.ts:
```diff
@@ -91,4 +94,29 @@ export class UserService {
     return movie;
   }
+
+  async getFavMovies({
```
А почему это юзер сервис? Какие действия с юзером здесь происходят?

Isidora25 commented on src/user/user.service.ts:
```diff
@@ -91,4 +94,29 @@ export class UserService {
+    const user = await this.getUserByLogin(login);
+
+    const movies = await this.favMoviesRepository.find({
+      where: { user: user },
```
where - лишний
Ты прямо по всем объекту юзера делаешь поиск? 
Не проще ли только по одному полю? например по id или email?

Isidora25 commented on src/user/user.service.ts:
```diff
@@ -91,4 +94,29 @@ export class UserService {
+    );
+
+    try {
+      return result;
```
а что ты тут проверяешь? где асинхронный вызов функции?

Isidora25 commented on src/user/user.service.ts:
```diff
@@ -91,4 +94,29 @@ export class UserService {
+        let moviesDetails = this.favMoviesService.getMoviesDetails(
+          movie.moviesId,
+        );
+        let movieIdData = lastValueFrom(moviesDetails.pipe(map((res) => res)));
```
Что делает pipe?

Mi4-04 committed:
add field watched in FavMovieDto

Mi4-04 committed:
fix comment

-move favMovies functions from user.service to fav-movies.service
-fix unnecessary

Isidora25 commented on src/fav-movies/fav-movies.module.ts:
```diff
@@ -4,11 +4,16 @@ import { FavMoviesEntity } from './fav-movies.entity';
+  imports: [
+    TypeOrmModule.forFeature([FavMoviesEntity, UserEntity]),
+    HttpModule,
+    UsersModule,
```
А модуль зачем?

Mi4-04 committed:
delete UserModule in FavMoviesModule

Isidora25 commented on src/fav-movies/fav-movies.module.ts:
```diff
@@ -12,7 +12,6 @@ import { UserService } from 'src/user/user.service';
   imports: [
     TypeOrmModule.forFeature([FavMoviesEntity, UserEntity]),
     HttpModule,
-    UsersModule,
```
А импорт модуля кому оставил?

Mi4-04 committed:
delete import UserModule

reshetov commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -34,4 +48,66 @@ export class FavMoviesService {
+    const user = await this.userService.getUserByLogin(login);
+
+    const favMoviesEntity: FavMoviesEntity = new FavMoviesEntity();
+    favMoviesEntity.moviesId = id;
```
А точно нужно каждый раз по одному атрибуту менять? (это я к тому что если у тебя модель из 30 аттрибутов то код будет в виде портянки. Попробуй переделать на сразу массовое присвоение

reshetov commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -34,4 +48,66 @@ export class FavMoviesService {
+      await favMoviesEntity.save();
+      return favMoviesEntity;
+    } catch (e) {
+      throw new InternalServerErrorException();
```
Чет так себе обработка ошибки - если что-то пошло не так перехватить ошибку и сгенерировать такую-же только новую)

reshetov commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -34,4 +48,66 @@ export class FavMoviesService {
+      await movie.save();
+      return movie;
+    } catch (e) {
+      throw new InternalServerErrorException();
```
см. выше

reshetov commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -34,4 +48,66 @@ export class FavMoviesService {
+    const movie = await this.favMoviesRepository.findOne({ moviesId });
+
+    if (!movie) {
+      throw new BadRequestException('Movie not found');
```
У тебя есть локали тут?

reshetov commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -34,4 +48,66 @@ export class FavMoviesService {
+    }
+
+    await this.favMoviesRepository.delete(movie.id);
+
```
Чисто оформительское - лишняя строка

Mi4-04 committed:
fix comment

-fix unnecessary
-fix addFavMovies

Isidora25 commented on src/fav-movies/fav-movies.service.ts:
```diff
@@ -55,16 +56,23 @@ export class FavMoviesService {
-    favMoviesEntity.moviesId = id;
-    favMoviesEntity.watched = false;
-    favMoviesEntity.user = user;
+    const favMoviesEntity = this.favMoviesRepository.manager.create(
```
Используй сразу save

Mi4-04 committed:
change create on save

Mi4-04 committed:
delete manager

Mi4-04 committed:
change year type
